docs/zq/language.md

```diff
@@ -1,341 +1,1969 @@
+the unicode code point at that offset in the string.
+
+If the `<value>` expression is type bytes, then the `<index>` operand
+must be coercible to an integer and the result is a unsigned 8-bit integer
+representing the byte value at that offset in the bytes sequence.
+
+### 6.7 Slices
+
+The slice operation can be applied to various data types and has the form:
+```
+<value> [ <from> : <to> ]
+```
+The `<from>` and `<to>` terms must be expressions that are coercible
+to integer and represent a range of index values to form a subset of elements
+from the `<value>` term provided.  The range begins at the `<from>` position
+and ends one before the `<to>` position.  A negative
+value of `<from>` or `<to>` represents a position relative to the
+end of the value being sliced.
+
+If the `<value>` expression is an array, then the result is an array of
+elements comprising the indicated range.
+
+If the `<value>` expression is an set, then the result is a set of
+elements comprising the indicated range ordered by total order of Zed values.
+
+If the `<value>` expression is an string, then the result is a substring
+consisting of unicode code points comprising the given range.
+
+If the `<value>` expression is type bytes, then the result is a bytes sequence
+consisting of bytes comprising the given range.
+
+### 6.8 Conditional
+
+A conditional expression has the form
+```
+<boolean> ? <expr> : <expr>`
+```
+The `<boolean>` expression is evaluated and must be result it type `bool`.
+If not, an error results.
+
+If the result is true, then the first `<expr>` expression is evaluated and becomes
+the result.  Otherwise, the second `<expr>` expression is evaluated.
+
+Note that if the expression has side effects,
+as with [aggregation calls](#aggregation-calls), only the selected expression
+will be evaluated.
+
+For example,
+```mdtest-command
+echo '{s:"foo",v:1}{s:"bar",v:2}' | zq -z 'yield (s=="foo") ? v : -v' -
+```
+produces
+```mdtest-output
+1
+-2
+```
+
+### 6.9 Function Calls
+
+Functions perform stateless transformations of their input value to their
+return value and utilize call-by value semantics with positional and unnamed
+arguments.  Some functions take a variable number of arguments.
+
+> The only available functions are built-in but user-defined functions and
+> library package management will be added to the Zed language soon.
+
+For example,
+```mdtest-command
+zq -z 'yield pow(2,3), lower("ABC")+upper("def"), typeof(1)'
+```
+produces
+```mdtest-output
+8.
+"abcDEF"
+<int64>
+```
+
+### 6.10 Aggregate Function Calls
+
+[Aggregate functions](reference.md#aggregates) may be called within an expression.
+Unlike the aggregation context provided by a summarizing group-by, such calls
+in expression context yield an output value for each input value.
+
+Note that because aggregate functions carry state which is typically dependent
+dependent on the order of input values, their use can prevent the runtime
```

Comment:
```suggestion
Note that because aggregate functions carry state which is typically
dependent on the order of input values, their use can prevent the runtime
```